sdk/python/kfp/compiler/compiler.py

```diff
@@ -920,6 +922,8 @@ def _create_workflow(
                 default=default_param_values[param.name]))
 
     op_transformers = [add_pod_env]
+    pod_labels = {_SDK_VERSION_LABEL: kfp.__version__, _SDK_ENV_LABEL:_SDK_ENV_DEFAULT}
```

Comment:
Can we add a test that verifies this?

Reply:
I once added a UT but that UT will break every time when we release a KFP SDK since we will need to verify the version(Therefore Alexey suggests we remove that UT).  I'm ok to add it back.

Reply:
Only one UT doesn't remove the metadata, I think the workload is ok to update just one UT when we release KFP SDK

Reply:
Added the UT back

Reply:
I modified the UT so that we don't need to upload the UT when we release KFP SDK